src/extension.ts

```diff
@@ -130,8 +131,15 @@ function displayLog(outputPanel: vscode.OutputChannel, message: string, show: bo
 }
 
 function executeCommand(command: (...args: any[]) => Promise<any>, thisArg: any, ...params: any[]) {
+    const errorMessage = params[params.length - 1];
     return command.call(thisArg, ...params).catch((err: string | Error) => {
         const error = typeof err === 'string' ? new Error(err) : err;
-        vscode.window.showErrorMessage(error.message);
+        let msg = error.message;
```

Comment:
well this PR just improved the way the error messages are built, trying to make them nicely. 
It was already assumed in the current implementation that error had a 'message' property. Are there any other kind of response we can receive from the server that are different from a string or an error?

Reply:
I dont know if such responses exist, I just know that it would blow up in this case. In a typed language you'd normally make sure your cast is ok, wont blow up.

Reply:
So was is the original assumption: 
You can get to this clause by either rejecting a promise, or throwing an error.
 -  Throw clauses only take Error objects as arguments, so the only way to have an empty error message is to create a ```new Error()``` error without one
 - Rejecting can be done with pretty much anything a promise can return, but honestly, we've never used anything but strings